modules/standard/Types.chpl

```diff
@@ -711,4 +711,110 @@ inline proc _bor_id(type t) return 0:t;
 pragma "no doc"
 inline proc _bxor_id(type t) return 0:t;
 
+
+/* Return the number of fields in a class or record as a param.
+   The count of fields includes types and param fields.
+ */
+proc numFields(type t) param
+  return __primitive("num fields", t);
+
+/* Get the name of the `i`th field in a class or record.
+   `i` should be in 1..numFields(t).
+
+   :arg t: a class or record type
+   :arg i: which field to get the name of
+   :returns: the name of the field, as a param string
+ */
+proc getFieldName(type t, param i:int) param
+  return __primitive("field num to name", t, i);
+
+pragma "no doc"
+proc fieldNumToName(type t, param i:int) param {
+  compilerError("fieldNumToName deprecated. Use getFieldName");
+}
```

Comment:
Is this here due to the fact that we had the routines in an undocumented form in the internal modules before, just so someone wouldn't use them accidentally?  Would you add a future which calls this routine to the test/deprecated directory so that we remember to pull this overload after 1.13 goes out?